[MacOS] Building the latest Development branch fails on Macbook Pro 2019

**Describe the bug**
Building `--target Editor` on macOS fails. This issue is the same as https://github.com/o3de/o3de/issues/14166 but occurs on a different machine. The following error appears:
```
CompileC /Users/macpro14/github/o3de/osx_build/External/RHI-3339ee5e/Code/O3DE.build/profile/Atom_RHI.Public.build/Objects-normal/x86_64/unity_0_cxx.o /Users/macpro14/github/o3de/osx_build/External/RHI-3339ee5e/Code/CMakeFiles/Atom_RHI.Public.dir/Unity/unity_0_cxx.cxx normal x86_64 c++ com.apple.compilers.llvm.clang.1_0.compiler (in target 'Atom_RHI.Public' from project 'O3DE')
    cd /Users/macpro14/github/o3de
    /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang -x c++ -target x86_64-apple-macos11.0 -fmessage-length\=0 -fdiagnostics-show-note-include-stack -fmacro-backtrace-limit\=0 -Wno-trigraphs -fpascal-strings -O2 -Wno-missing-field-initializers -Wno-missing-prototypes -Wno-return-type -Wno-non-virtual-dtor -Wno-overloaded-virtual -Wno-exit-time-destructors -Wno-missing-braces -Wparentheses -Wswitch -Wno-unused-function -Wno-unused-label -Wno-unused-parameter -Wno-unused-variable -Wunused-value -Wno-empty-body -Wno-uninitialized -Wno-unknown-pragmas -Wno-shadow -Wno-four-char-constants -Wno-conversion -Wno-constant-conversion -Wno-int-conversion -Wno-bool-conversion -Wno-enum-conversion -Wno-float-conversion -Wno-non-literal-null-conversion -Wno-objc-literal-conversion -Wno-shorten-64-to-32 -Wno-newline-eof -Wno-c++11-extensions -DCMAKE_INTDIR\=\"profile\" -D_HAS_EXCEPTIONS\=0 -D_PROFILE -DAZ_PROFILE_BUILD -DNDEBUG -DAZ_ENABLE_TRACING -DAZ_ENABLE_DEBUG_TOOLS -DAZ_BUILD_CONFIGURATION_TYPE\=\"profile\" -D_FORTIFY_SOURCE\=2 -DAPPLE -DMAC -D__APPLE__ -DDARWIN -DO3DE_STACK_CAPTURE_DEPTH\=3 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX12.1.sdk -fasm-blocks -fstrict-aliasing -Wdeprecated-declarations -Winvalid-offsetof -g -Wno-sign-conversion -Wno-infinite-recursion -Wno-move -Wno-comma -Wno-block-capture-autoreleasing -Wno-strict-prototypes -Wno-range-loop-analysis -Wno-semicolon-before-method-body -I/Users/macpro14/github/o3de/osx_build/lib/profile/include -I/Users/macpro14/github/o3de/Gems/Atom/RHI/Code/Source -I/Users/macpro14/github/o3de/Gems/Atom/RHI/Code/Source/Platform/Mac -I/Users/macpro14/github/o3de/Gems/Atom/RHI/Code/Include -I/Users/macpro14/github/o3de/Code/Framework/AzCore -I/Users/macpro14/github/o3de/Code/Framework/AzCore/Platform/Mac -I/Users/macpro14/github/o3de/Code/Framework/AzCore/Platform/Common -I/Users/macpro14/github/o3de/Code/Framework/AzFramework -I/Users/macpro14/github/o3de/Code/Framework/AzFramework/Platform/Mac -I/Users/macpro14/github/o3de/Code/Framework/AzNetworking -I/Users/macpro14/github/o3de/Code/Framework/AzNetworking/Platform/Common -I/Users/macpro14/github/o3de/Code/Framework/AzNetworking/Platform/Mac -I/Users/macpro14/github/o3de/osx_build/Code/Framework/AzNetworking/Azcg/Generated/AzNetworking -I/Users/macpro14/github/o3de/Code/Framework/AtomCore -isystem /Users/macpro14/github/o3de/3rdParty/packages/Lua-5.4.4-rev1-mac/Lua/include -isystem /Users/macpro14/github/o3de/3rdParty/packages/RapidJSON-1.1.0-rev1-multiplatform/RapidJSON/include -isystem /Users/macpro14/github/o3de/3rdParty/packages/RapidXML-1.13-rev1-multiplatform/RapidXML/include -isystem /Users/macpro14/github/o3de/3rdParty/packages/zlib-1.2.11-rev5-mac/zlib/include -isystem /Users/macpro14/github/o3de/3rdParty/packages/zstd-1.35-multiplatform/zstd/lib -isystem /Users/macpro14/github/o3de/3rdParty/packages/cityhash-1.1-multiplatform/cityhash/src -isystem /Users/macpro14/github/o3de/3rdParty/packages/lz4-1.9.3-vcpkg-rev4-mac/lz4/lz4/include -I/Users/macpro14/github/o3de/osx_build/External/RHI-3339ee5e/Code/O3DE.build/profile/Atom_RHI.Public.build/DerivedSources-normal/x86_64 -I/Users/macpro14/github/o3de/osx_build/External/RHI-3339ee5e/Code/O3DE.build/profile/Atom_RHI.Public.build/DerivedSources/x86_64 -I/Users/macpro14/github/o3de/osx_build/External/RHI-3339ee5e/Code/O3DE.build/profile/Atom_RHI.Public.build/DerivedSources -F/Users/macpro14/github/o3de/osx_build/lib/profile -fno-exceptions -fvisibility\=hidden -fvisibility-inlines-hidden -Wall -Werror -Wno-inconsistent-missing-override -Wrange-loop-analysis -Wno-unknown-warning-option -Wno-parentheses -Wno-reorder -Wno-switch -Wno-undefined-var-template -fstack-protector-all -fstack-check -fPIC -std\=c++17 -MMD -MT dependencies -MF /Users/macpro14/github/o3de/osx_build/External/RHI-3339ee5e/Code/O3DE.build/profile/Atom_RHI.Public.build/Objects-normal/x86_64/unity_0_cxx.d --serialize-diagnostics /Users/macpro14/github/o3de/osx_build/External/RHI-3339ee5e/Code/O3DE.build/profile/Atom_RHI.Public.build/Objects-normal/x86_64/unity_0_cxx.dia -c /Users/macpro14/github/o3de/osx_build/External/RHI-3339ee5e/Code/CMakeFiles/Atom_RHI.Public.dir/Unity/unity_0_cxx.cxx -o /Users/macpro14/github/o3de/osx_build/External/RHI-3339ee5e/Code/O3DE.build/profile/Atom_RHI.Public.build/Objects-normal/x86_64/unity_0_cxx.o

** BUILD FAILED **


The following build commands failed:
	CompileC /Users/macpro14/github/o3de/osx_build/External/Metal-69b2f455/Code/O3DE.build/profile/Atom_RHI_Metal.Private.Static.build/Objects-normal/x86_64/unity_1_cxx.o /Users/macpro14/github/o3de/osx_build/External/Metal-69b2f455/Code/CMakeFiles/Atom_RHI_Metal.Private.Static.dir/Unity/unity_1_cxx.cxx normal x86_64 objective-c++ com.apple.compilers.llvm.clang.1_0.compiler (in target 'Atom_RHI_Metal.Private.Static' from project 'O3DE')
(1 failure)
Elapsed time: 33 s. (time), 0.003341 s. (clock)
macpro14@macpro14s-macbook-pro osx_build %
``` 

The issue has been confirmed on the following machine:
Macbook Pro 2019 Big Sur 11.6.1  XCode version 13.2.1
but it also occurred before on other machines in earlier commit: https://github.com/o3de/o3de/issues/14166

This issue occurs when building AutomatedTesting.

Please refer to the attached building logs for more information.

**Steps to reproduce**

1. Create new or move your preexisting 3rdParty folder containing CMake into the o3de folder.
2. Obtain python packages with get_python.sh script located in the o3de/python directory.
3. Create osx_build folder under the o3de location.
4. Set the following temporary environment variables in the Terminal:
```
export PATH=$PATH:/Users/<home_folder_name>/github/o3de/3rdParty/CMake/CMake.app/Contents/bin
export LY_3RDPARTY_PATH=/Users/<home_folder_name>/github/o3de/3rdParty
```
5. Configure and generate the Xcode solution using this command:
`cmake .. -G "Xcode" -DLY_3RDPARTY_PATH=$LY_3RDPARTY_PATH -DLY_PROJECTS=AutomatedTesting -DLY_UNITY_BUILD=1 -DAZ_USE_PHYSX5=ON`
6. Build Mac binaries:
`cmake -E time cmake --build . --target ALL_BUILD --config profile -j 4`

**Expected behavior**
Building completes successfully.

**Actual behavior**
Building fails due to a command failure.

**Found in Branch**
Development

Commit ID from [o3de/o3de](https://github.com/o3de/o3de) Repository
f829a06

**Device :**
Device: Macbook Pro 2019
OS: macOS
Version: Big Sur 11.6.1
CPU: Intel i9
GPU: AMD Radeon Pro 5500M 4 GB
Memory: 16GB

**Additional context**
[building_fail_f829a06.txt](https://github.com/o3de/o3de/files/10732393/building_fail_f829a06.txt)

@moudgils looking at the attached log the error is due to the Metal Device features structure not having an `m_customResolvePositions` member variable.

```c++
In file included from /Users/macpro14/github/o3de/osx_build/External/Metal-69b2f455/Code/CMakeFiles/Atom_RHI_Metal.Private.Static.dir/Unity/unity_1_cxx.cxx:11:
/Users/macpro14/github/o3de/Gems/Atom/RHI/Metal/Code/Source/RHI/Device.cpp:347:24: error: no member named 'm_customResolvePositions' in 'AZ::RHI::DeviceFeatures'; did you mean 'm_customSamplePositions'?
            m_features.m_customResolvePositions = m_metalDevice.programmableSamplePositionsSupported;
                       ^~~~~~~~~~~~~~~~~~~~~~~~
                       m_customSamplePositions
In file included from /Users/macpro14/github/o3de/osx_build/External/Metal-69b2f455/Code/CMakeFiles/Atom_RHI_Metal.Private.Static.dir/Unity/unity_1_cxx.cxx:3:
In file included from /Users/macpro14/github/o3de/Gems/Atom/RHI/Metal/Code/Source/RHI/AsyncUploadQueue.cpp:11:
In file included from /Users/macpro14/github/o3de/Gems/Atom/RHI/Code/Include/Atom/RHI/BufferPool.h:11:
In file included from /Users/macpro14/github/o3de/Gems/Atom/RHI/Code/Include/Atom/RHI/BufferPoolBase.h:10:
In file included from /Users/macpro14/github/o3de/Gems/Atom/RHI/Code/Include/Atom/RHI/ResourcePool.h:11:
In file included from /Users/macpro14/github/o3de/Gems/Atom/RHI/Code/Include/Atom/RHI/DeviceObject.h:10:
In file included from /Users/macpro14/github/o3de/Gems/Atom/RHI/Code/Include/Atom/RHI/Device.h:13:
/Users/macpro14/github/o3de/Gems/Atom/RHI/Code/Include/Atom/RHI.Reflect/DeviceFeatures.h:41:18: note: 'm_customSamplePositions' declared here
            bool m_customSamplePositions = false;
```

--- a/Code/Framework/AzToolsFramework/AzToolsFramework/Manipulators/EditorVertexSelection.cpp
+++ b/Code/Framework/AzToolsFramework/AzToolsFramework/Manipulators/EditorVertexSelection.cpp
@@ -1084,7 +1084,7 @@ namespace AzToolsFramework
     {
         if (EditorVertexSelectionBase<Vertex>::m_translationManipulator)
         {
-            return EditorVertexSelectionBase<Vertex>::m_translationManipulator->m_vertices.size();
+            return aznumeric_cast<int>(EditorVertexSelectionBase<Vertex>::m_translationManipulator->m_vertices.size());
         }
 
         return 0;

--- a/Gems/Atom/RHI/Metal/Code/Source/RHI/Device.cpp
+++ b/Gems/Atom/RHI/Metal/Code/Source/RHI/Device.cpp
@@ -344,7 +344,7 @@ namespace AZ
             m_features.m_computeShader = true;
             m_features.m_independentBlend = true;
             m_features.m_dualSourceBlending = true;
-            m_features.m_customResolvePositions = m_metalDevice.programmableSamplePositionsSupported;
+            m_features.m_customSamplePositions = m_metalDevice.programmableSamplePositionsSupported;
             m_features.m_indirectDrawSupport = false;
             
             //Metal drivers save and load serialized PipelineLibrary internally

